source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,14 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint endpoint(endpoint_.value());
```

Comment:
I cannot have one call the other without using `const_cast`. Are you okay is this use of `const_cast`?